apps/webapp/src/components/swap/unclaimed-swaps.tsx

```diff
@@ -0,0 +1,75 @@
+import { Button, Card } from '@penumbra-zone/ui';
+import { useLoaderData, useRevalidator } from 'react-router-dom';
+import { SwapLoaderResponse } from './swap-loader.tsx';
+import { AssetIcon } from '../shared/asset-icon.tsx';
+import { issueSwapClaim } from '../../state/swap.ts';
+import { SwapRecord } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { getSwapRecordCommitment } from '@penumbra-zone/types';
+import { toast, useToast } from '@penumbra-zone/ui/components/ui/use-toast.ts';
+import { errorTxToast, loadingTxToast, successTxToast } from '../shared/toast-content.tsx';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+const claimSwap = async (
+  swap: SwapRecord,
+  toastFn: typeof toast,
+  setInProgress: Dispatch<SetStateAction<boolean>>,
+  reloadData: () => void, // Used to refresh page state after action
+) => {
+  const { dismiss } = toastFn(loadingTxToast);
+  setInProgress(true);
+
+  try {
+    const commitment = getSwapRecordCommitment(swap);
+    const swapClaimTxHash = await issueSwapClaim(commitment);
+
+    dismiss();
+    toastFn(successTxToast(swapClaimTxHash));
+  } catch (e) {
+    toastFn(errorTxToast(e));
+    throw e;
+  } finally {
+    dismiss();
+    setInProgress(false);
+    reloadData();
+  }
+};
+
+export const UnclaimedSwaps = () => {
+  const { toast } = useToast();
+  const { revalidate } = useRevalidator();
+  const [inProgress, setInProgress] = useState(false);
+  const { unclaimedSwaps } = useLoaderData() as SwapLoaderResponse;
+  if (!unclaimedSwaps.length) return <></>;
+
+  return (
+    <Card className=''>
+      <p className='bg-text-linear bg-clip-text font-headline text-xl font-semibold leading-[30px] text-transparent md:text-2xl md:font-bold md:leading-9'>
+        Unclaimed Swaps
+      </p>
+      <p className='text-gray-400'>
+        Swaps on Penumbra are a two step process. The first transaction issues the request and the
+        second claims the result of the swap action. For some reason, these second transactions were
+        not completed. Claim them!
+      </p>
+      {unclaimedSwaps.map(({ swap, asset1, asset2 }, i) => (
+        <div key={i} className='mt-4 flex justify-between'>
```

Comment:
It's generally a no-no to use an array index as the React key, as this can lead to rendering bugs. Could you try encoding the swap commitment to base64 and using that?